tools/capable.py

```diff
@@ -128,30 +163,47 @@
 
 class Data(ct.Structure):
     _fields_ = [
-        ("pid", ct.c_ulonglong),
-        ("uid", ct.c_ulonglong),
+        ("pid_tgid", ct.c_uint64),
+        ("tgid", ct.c_uint32),
+        ("pid", ct.c_uint32),
+        ("uid", ct.c_uint32),
         ("cap", ct.c_int),
         ("audit", ct.c_int),
-        ("comm", ct.c_char * TASK_COMM_LEN)
-    ]
+        ("comm", ct.c_char * TASK_COMM_LEN),
+    ] + ([("kernel_stack_id", ct.c_int)] if args.kernel_stack else []) \
+      + ([("user_stack_id", ct.c_int)] if args.user_stack else [])
 
 # header
-print("%-9s %-6s %-6s %-16s %-4s %-20s %s" % (
-    "TIME", "UID", "PID", "COMM", "CAP", "NAME", "AUDIT"))
+print("%-9s %-6s %-6s %-6s %-16s %-4s %-20s %s" % (
+    "TIME", "UID", "PID", "TID", "COMM", "CAP", "NAME", "AUDIT"))
+
+def print_stack(bpf, stack_id, tgid):
+    if stack_id < 0:
+        print("        %d" % stack_id)
```

Comment:
Don't just print the stack_id which is actually an error code which by itself is not apparent to user what is going on. How about print something like `[Missed User Stack]` or `[Missed Kernel Stack]` similar to `offwaketime.py`?

Reply:
I like it better. I'll do this change and post a new pull request.
Thanks for the review!